packages/block-editor/src/components/block-draggable/index.js

```diff
@@ -67,14 +67,18 @@ const BlockDraggable = ( {
 			__experimentalTransferDataType="wp-blocks"
 			transferData={ transferData }
 			onDragStart={ ( event ) => {
-				startDraggingBlocks( clientIds );
-				isDragging.current = true;
+				// Defer hiding the dragged source element to the next
+				// frame to enable dragging.
+				window.requestAnimationFrame( () => {
```

Comment:
This is to retain the former behavior of asynchronous `onDragStart` before. I also changed `setTimeout` to `requestAnimationFrame` to better describe the intent.